src/indicators/drought/wrsi.py

```diff
@@ -0,0 +1,382 @@
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import pandas as pd
+import rioxarray  # noqa: F401
+import xarray as xr
+from rasterio.crs import CRS
+
+logger = logging.getLogger(__name__)
+
+_BASE_URL = (
+    "https://edcftp.cr.usgs.gov/project/fews/"
+    "africa/west/dekadal/wrsi-chirps-etos/{region}/"
+)
+
+_URL_ZIP_FILENAME = "w{year:04}{dekad}{region}.zip"
+
+_BASE_FILENAME = "w{year:04}{dekad}{type}.tif"
+
+_RAW_DIR = Path(
+    os.getenv("AA_DATA_DIR"), "public", "raw", "glb", "wrsi", "west_africa"
+)
+
+_PROCESSED_FILENAME = "wrsi_{region}_{type}.nc"
+
+_PROCESSED_DIR = Path(
+    os.getenv("AA_DATA_DIR"), "public", "processed", "glb", "wrsi"
+)
+
+# defined here: https://gis.stackexchange.com/questions/
+# 177447/defining-the-correct-aea-proj4string-for-
+# fewsnet-rainfall-data-southern-africa
+_WRSI_CRS = CRS.from_proj4(
+    "+proj=aea +lat_1=-19.0 +lat_2=21.0 +lat_0=1.0 "
+    "+lon_0=20 +x_0=0 +y_0=0 +ellps=clrk66 +units=m +no_defs"
+)
+
+RegionArgument = Literal["cropland", "rangeland"]
+_VALID_REGION = get_args(RegionArgument)
+
+
+def download_wrsi(
+    region: RegionArgument,
+    start_date: Union[date, str, List[int], None] = None,
+    end_date: Union[date, str, List[int], None] = None,
+    clobber: bool = False,
+):
+    """Download USGS WRSI data
+
+    Downloads USGS WRSI data from their website.
+    Currently only downloads data for west African
+    croplands and rangelands, where data is available
+    from 2001, for all dekads between 13 and 33 of
+    each year.
+    Parameters
+    ----------
+    region: RegionArgument
+        One of 'cropland' or 'rangeland'.
+    start_date: Union[date, str, List[int], None]
+        Starting date for data downloading passed in as
+        a ``datetime.date``, ISO8601 date string, or
+        ``list`` of ``int`` for year and dekad. If ``None``,
+        defaults to ``[2001, 13]``.
+    end_date: Union[date, str, List[int], None]
+        Ending date for data downloading passed in as
+        a ``datetime.date``, ISO8601 date string, or
+        ``list`` of ``int`` for year and dekad. If ``None``,
+        defaults to ``datetime.date.today()``.
+    clobber: bool (optional)
+        If ``True``, rewrite existing files. Otherwise,
+        only download files not alreayd available.
+
+    Returns
+    -------
+        None
+    """
+    if region not in _VALID_REGION:
+        raise ValueError("`region` must be one of 'cropland' or 'rangeland'.")
+
+    if start_date is None:
+        start_year, start_dekad = 2001, 13
+    elif isinstance(start_date, list):
+        start_year, start_dekad = start_date[0], start_date[1]
+    else:
+        start_year, start_dekad = _date_to_dekad(start_date)
+
+    if isinstance(end_date, list):
+        end_year, end_dekad = end_date[0], end_date[1]
+    else:
+        if end_date is None:
+            end_date = date.today()
+        end_year, end_dekad = _date_to_dekad(end_date)
+
+    i_year = max(start_year, 2001)
```

Comment:
I would move this check already up to the `start_year,start_dekad` determination and thus add an extra line with `start_year = max(start_year,2001)`